File: MPI/Fortran/02 Counts.f90
PROGRAM main
  USE MPI
  IMPLICIT NONE
  INTEGER :: errcode
  INTEGER :: rank, nproc
  CALL MPI_Init(errcode)
  CALL MPI_Comm_size(MPI_COMM_WORLD, nproc, errcode)
  CALL MPI_Comm_rank(MPI_COMM_WORLD, rank, errcode)
  PRINT '(A, I3, A, I3)', 'I am processor ', rank + 1, ' of ', nproc
  CALL MPI_Finalize(errcode)
END PROGRAM main

File: MPI/C/02 Counts.c
#include<stdio.h>
#include<mpi.h>

int main(int argc, char** argv)
{
  int rank, nproc;
  MPI_Init(&argc, &argv);
  MPI_Comm_size(MPI_COMM_WORLD, &nproc);
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);

  printf("I am processor %i of %i\n", rank+1, nproc);
  MPI_Finalize();
}
